### DISCUSSION TITLE
Real Rust code (simple projects)

### DISCUSSION THREAD

**Discussion by philberty** [Rust-GCC/gccrs#713] [Q&A]
We are aiming to try and compile real rust code soon see: https://github.com/Rust-GCC/gccrs/issues/682. The goal here is not that we expect the projects compile correctly first go, but we wish to find as many bugs as possible. This will help me figure out what gaps the compiler has for planning purposes.

We currently don't support macros and do not have a compiled libcore, liballoc etc yet so these projects should be simple crates.

**Comment by bjorn3** [comment]
Maybe the baseline implementation in crc32fast? This crate is used by object for crc32 calculations.

**Comment by CAD97** [comment]
I'd like to offer a few of my own crates,

- [erasable](https://lib.rs/crates/erasable), a set of traits for types/pointers reconstructable from `*mut ()`;
- [ptr-union](https://lib.rs/crates/ptr-union), utilities for alignment-tagging erasable pointers; and
- [slice-dst](https://lib.rs/crates/slice-dst), utilities for supporting custom slice-tailed DSTs.

While they'd likely take some preprocessing to serve as good test cases (to remove libcore, liballoc, macros), they do not rely on libcore to provide any of their core functionality, but just for the purpose of attaching the functionality to libcore/liballoc. If I recall correctly, `core::{marker::PhantomData, mem::{MaybeUninit, ManuallyDrop}}` are the only lang features they really rely on being available.

**Reply by CAD97** [reply]
And [here's a pre-stripped version of erasable](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=57c7faf0057a348d7e4fcdb3b7c607f5).
